libmscore/element.cpp

```diff
@@ -177,6 +177,7 @@ Element::Element(const Element& e)
     : ScoreElement(e)
 {
     _parent     = e._parent;
+    _albumParent = e._albumParent;
```

Comment:
What is the purpose of the new `_albumParent` member of  `Element`? And, how does it differ from `_parent`?

Reply:
This is used to fix the teleporting `ScoreView` problem. When `layout` is called on a score it deletes and re-adds systems and pages. When `adjustCanvasPosition` is called (while in album-mode) the parent-child relationships of pages-systems-measures might be the ones we are seeing on screen or (if Score::layout has been called for an individual score) the ones that are present if the Score is alone (not part of an album). _albumParent saves the parent system and page when `layout` is called for the Temporary Album Score so that we can consistently get the true position of each element in the. 

It's pretty rough and needs some cleaning up.